### PR TITLE
Add CI build

### PR DESCRIPTION
**Pull request by r-ash**
Builds on #50 and should be reviewed after that

This PR adds a CI build for python project which runs on windows, linux and mac. The build runs tests, coverage and lint checks. Addresses #51

### CODE REVIEW

**Review by rlglaubius — APPROVED** [review]
Looks good to me, but I rate my confidence as "low" in this space.

File: .github/workflows/leapfrog-py-test.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - {os: macos-latest,   py: '3.12'}
+          - {os: ubuntu-latest,  py: '3.8'}
+          - {os: ubuntu-latest,  py: '3.10'}
+          - {os: ubuntu-latest,  py: '3.11'}
+          - {os: ubuntu-latest,  py: '3.12'}
+          - {os: windows-latest, py: '3.12'}
```
Comment (M-Kusumgar) [inline .github/workflows/leapfrog-py-test.yaml:29]: why not 3.9?

File: .github/workflows/leapfrog-py-test.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+      - name: Test
+        run: |
+          hatch run cov-ci
+
+      - name: Lint
+        run: |
+          # https://github.com/python/mypy/issues/10863
+          mkdir -p .mypy_cache
+          hatch run lint:all
```
Comment (M-Kusumgar) [inline .github/workflows/leapfrog-py-test.yaml:73]: happy with this but might be nicer to run lint before tests, feel like its more immediate feedback

**Review by M-Kusumgar — APPROVED** [review]
Looks good to me, have some optional comments!

File: .github/workflows/leapfrog-py-test.yaml (hunk 1 above)
Comment (r-ash) [reply .github/workflows/leapfrog-py-test.yaml]: Ah just missed that when I went through and updated the versions. Testing 3.9 - 3.12 which I think should be enough.

File: .github/workflows/leapfrog-py-test.yaml (hunk 2 above)
Comment (r-ash) [reply .github/workflows/leapfrog-py-test.yaml:73]: Yeah makes sense, tests are actually faster than lint atm but will keep in mind as that probably won't always be true

Comment (M-Kusumgar) [reply .github/workflows/leapfrog-py-test.yaml:73]: damn tests being faster than lint is amazing

**Pull request merged**